scene/2d/audio_stream_player_2d.cpp

```diff
@@ -254,14 +253,15 @@ void AudioStreamPlayer2D::set_stream(Ref<AudioStream> p_stream) {
 		setseek = -1;
 	}
 
-	stream = p_stream;
-	stream_playback = p_stream->instance_playback();
+	if (p_stream.is_valid()) {
+		stream = p_stream;
+		stream_playback = p_stream->instance_playback();
+	}
 
 	AudioServer::get_singleton()->unlock();
 
 	if (stream_playback.is_null()) {
 		stream.unref();
```

Comment:
Isn't this `unref`ing the stream twice if the new stream is null?

Reply:
Hmm so do you think
```
if (stream_playback.is_null()) {
 		stream.unref();
```
can be removed?

It was there in the old code in all versions (AudioStreamPlayer and AudioStreamPlayer2D/3D had similar but different `set_stream` methods) so I didn't think to remove it.

Reply:
I see, I will test it and update the pull request